libvast/src/system/partition.cpp

```diff
@@ -735,10 +741,12 @@ partition_actor::behavior_type passive_partition(
   partition_actor::stateful_pointer<passive_partition_state> self, uuid id,
   store_actor legacy_archive, filesystem_actor filesystem,
   const std::filesystem::path& path) {
+  auto id_string = to_string(id);
   self->state.self = self;
   self->state.path = path;
   self->state.archive = legacy_archive;
-  auto id_string = to_string(id);
+  self->state.id = id;
+  self->state.name = "partition-" + id_string;
```

Comment:
How about just this and ditch the extra local variable?

```cpp
  self->state.id = to_string(id);
  self->state.name = "partition-" + self->state.name;
```

Reply:
We're also using the `id_string` for the tracepoint a few lines below.